controllers/flinkcluster_reconciler.go

```diff
@@ -373,12 +420,12 @@ func (reconciler *ClusterReconciler) reconcileJob() (ctrl.Result, error) {
 	}
 
 	// Update or restart
-	var jobID = reconciler.getFlinkJobID()
 	if desiredJob != nil && observedJob != nil {
+		var jobID = reconciler.getFlinkJobID()
 		var restartPolicy = observed.cluster.Spec.Job.RestartPolicy
 		var observedJobStatus = observed.cluster.Status.Components.Job
 
-		if shouldRestartJob(restartPolicy, observedJobStatus) {
+		if shouldRestartJob(restartPolicy, observedJobStatus) || shouldUpdateJob(observed) {
```

Comment:
If we updated both components and job, we need to wait until the cluster is up and running before restarting the job.

Reply:
Fixed to check it by `isClusterReady`.